apps/designer/app/designer/shared/assets/use-serial-submit.ts

```diff
@@ -0,0 +1,41 @@
+import { useActionData, useSubmit, type SubmitOptions } from "@remix-run/react";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const initialSubmission: {
+  data: Array<[string, FormDataEntryValue]>;
+  options: SubmitOptions;
+} = {
+  data: [],
+  options: {},
+};
+
+/**
+ * Wrapps useSubmit hook to submit form data serially.
+ * This is useful when you want to submit multiple files, but send them one after another.
+ * Specifically Vercel has a single upload limit of 4.5MB, so we need to send them one by one so that this limit is per file, not per total upload
+ */
+export const useSerialSubmit = () => {
+  const submit = useSubmit();
```

Comment:
Not sure if this should be part of this PR, but in general I think `useFetcher` is much better for actions like uploading assets, or deleting a font — all "small" forms

Reply:
rewriting to useFetcher right now

Reply:
done